lib/Mongo/MongoCollection.php

```diff
@@ -290,13 +316,18 @@ public function insert(&$a, array $options = [])
     public function batchInsert(array &$a, array $options = [])
     {
         foreach ($a as $key => $item) {
+            $this->validateDocument($a[$key]);
             $this->ensureDocumentHasMongoId($a[$key]);
```

Comment:
I didn't check whether having one invalid document in a `batchInsert` call skips that invalid document or if the entire batch fails. Either way, we should ensure the same handling here.